python/GangaLHCb/Lib/GaudiRunApp/GaudiRunApp.py

```diff
@@ -0,0 +1,248 @@
+    myApp = GaudiRun()
+    myApp.directory = "$SOMEPATH/DaVinciDev_v40r2"
+    myApp.myOpts = "$SOMEPATH/DaVinciDev_v40r2/myDaVinciOpts.py"
+    j.application = myApp
+    j.submit()
+
+    To setup a minimal application you can also run the helper function:
+
+    prepare_cmake_app(myApp, myVer, myPath, myGetpack)
+
+    """
+    _schema = Schema(Version(1, 0), {
+        # Options created for constructing/submitting this app
+        'directory':    SimpleItem(preparable=1, defvalue=None, typelist=[None, str], comparable=1,
+            doc='A path to the project that you\'re wanting to run.'),
+        'build_opts':   SimpleItem(defvalue=[""], typelist=[str], sequence=1, strict_sequence=0,
+            doc="Options to be passed to 'make ganga-input-sandbox'"),
+        'myOpts':       GangaFileItem(defvalue=None, doc='File which contains the extra opts I want to pass to gaudirun.py'),
+        'uploadedInput': GangaFileItem(defvalue=None, doc='This stores the input for the job which has been pre-uploaded so that it gets to the WN'),
+
+        # Prepared job object
+        'is_prepared':  SimpleItem(defvalue=None, strict_sequence=0, visitable=1, copyable=1, hidden=0, typelist=[None, bool, ShareDir], protected=0, comparable=1,
+            doc='Location of shared resources. Presence of this attribute implies the application has been prepared.'),
+        'hash':         SimpleItem(defvalue=None, typelist=[None, str], hidden=0,
+            doc='MD5 hash of the string representation of applications preparable attributes'),
+        'arch':         SimpleItem(defvalue="x86_64-slc6-gcc49-opt", typelist=[str], doc='Arch the application was built for'),
+        })
+    _category = 'applications'
+    _name = 'GaudiRun'
+    _exportmethods = ['prepare', 'unprepare', 'exec_cmd', 'getDir']
+
+    def __init__(self):
+        super(GaudiRun, self).__init__()
+
+    def unprepare(self, force=False):
+        logger.debug('Running unprepare in GaudiRun app')
+        if self.is_prepared is not None:
+            self.decrementShareCounter(self.is_prepared.name)
+            self.is_prepared = None
+        self.hash = None
+        self.uploadedInput = None
+
+    def prepare(self, force=False):
+        """
+        This method creates a set of prepared files for the application to pass to the RTHandler
+        """
+
+        if (self.is_prepared is not None) and (force is not True):
+            raise ApplicationPrepareError('%s application has already been prepared. Use prepare(force=True) to prepare again.' % getName(self))
+
+        # lets use the same criteria as the configure() method for checking file existence & sanity
+        # this will bail us out of prepare if there's somthing odd with the job config - like the executable
+        # file is unspecified, has a space or is a relative path
+        self.configure(self)
+        logger.info('Preparing %s application.' % getName(self))
+        setattr(self, 'is_prepared', ShareDir())
```

Comment:
Copied from elsewhere. Should be fixed in Executable as well. Lots of bad code around the `is_prepared` which we've inherited